src/envoy/http/path_matcher/filter_config_test.cc

```diff
@@ -159,6 +158,49 @@ rules {
       ProtoValidationException, "Duplicated pattern");
 }
 
+TEST(FilterConfigTest, InvalidPattern) {
+  const char kFilterConfig[] = R"(
+rules {
+  operation: "1.cloudesf_testing_cloud_goog.Bar"
+  pattern {
+    http_method: "GET"
+    uri_template: "/bar/{id{x}}"
+  }
+})";
+
+  ::google::api::envoy::http::path_matcher::FilterConfig config_pb;
+  ASSERT_TRUE(TextFormat::ParseFromString(kFilterConfig, &config_pb));
+  ::testing::NiceMock<Server::Configuration::MockFactoryContext> mock_factory;
+
+  EXPECT_THROW_WITH_REGEX(
+      FilterConfig cfg(config_pb, EMPTY_STRING, mock_factory),
+      ProtoValidationException, "invalid pattern");
+}
+
+TEST(FilterConfigTest, NonStandardHttpMethod) {
+  const char kFilterConfigBasic[] = R"(
+rules {
+  operation: "1.cloudesf_testing_cloud_goog.Bar"
+  pattern {
+    http_method: "NonStandardMethod"
```

Comment:
we do allow custom []([method](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L372).)